### PR TITLE
Update Helm chart to support GCP MP Enterprise integration

### PR DESCRIPTION
**Pull request by linhlam-kc**
## What does this PR change?

Add necessary changes to support Kubecost Enterprise product listed on the Google Cloud Marketplace. So, it can help to accelerate the update process for this integration.

## Does this PR rely on any other PRs?

N/A

## How does this PR impact users? (This is the kind of thing that goes in release notes!)

This is required code changes for our product listed on GCP Marketplace to transact with customers using GCP MP Private Offers

## Links to Issues or ZD tickets this PR addresses or fixes

[2067](https://github.com/kubecost/cost-analyzer-helm-chart/issues/2067)

## How was this PR tested?

The integration was tested ad approved on the GCP MP at https://console.cloud.google.com/marketplace/product/kubecost-public/kubecost-ent 

## Have you made an update to documentation?

Yes, at the specific repo for GCP Marketplace https://github.com/kubecost/gcp-marketplace

### CODE REVIEW

File: cost-analyzer/values.yaml [hunk 1]
```diff
@@ -894,6 +898,11 @@ kubecostAdmissionController:
 costEventsAudit:
   enabled: false
 
+# Reporting Secret for GCP Marketplace integration.
+# The value will be replaced automatically when Kubecost is deployed following GCP MP instructions.
+# Do not change the value manually.
+reportingSecret: "kubecost-reporting-secret"
```
Comment (thomasvn) [inline cost-analyzer/values.yaml]: @linhlam-kc If this value shouldn't be changed manually, should we set this as a the default in `cost-analyzer/templates/gcpstore-config-map-template.yaml` and remove it from this `values.yaml`?

File: cost-analyzer/values.yaml [hunk 2]
```diff
@@ -22,6 +22,10 @@ global:
     proxy: true # If true, the kubecost frontend will route to your grafana through its service endpoint
 #    fqdn: cost-analyzer-grafana.default.svc
 
+  # Enable only when you are using GCP Marketplace ENT listing. Learn more at https://console.cloud.google.com/marketplace/product/kubecost-public/kubecost-ent
```
Comment (thomasvn) [inline cost-analyzer/values.yaml:25]: @linhlam-kc Could you provide more comment on what the purpose of the `ubbagent` is and why it is necessary when enabling Kubecost via the GCP Marketplace?

File: cost-analyzer/values.yaml (hunk 1 above)
Comment (linhlam-kc) [reply cost-analyzer/values.yaml]: It is required to be parameterized and to be in the Helm chart values file, so GCP MP deployment tool can substitute the value at the deployment time. GCP MP does not use Helm client to deploy Helm Chart in normal way.

Comment (thomasvn) [reply cost-analyzer/values.yaml]: Would it work if we removed these lines from `values.yaml`, and instead did something like this in the template:

```yaml
          - name: AGENT_ENCODED_KEY
            valueFrom:
              secretKeyRef:
                name: {{ default "kubecost-reporting-secret" .Values.reportingSecret }}
                key: reporting-key
```

Comment (linhlam-kc) [reply cost-analyzer/values.yaml]: Thank you Thomas ! It should work, I have tested locally and apply the changes in the latest commit [8285d6c](https://github.com/kubecost/cost-analyzer-helm-chart/pull/2157/commits/8285d6cbc83225ba14eb28271601519cb4b44170)

**Review by thomasvn — APPROVED** [review]

**Pull request merged**